modular_skyrat/modules/modular_implants/code/soulcatcher/handheld_soulcatcher.dm

```diff
@@ -47,68 +56,112 @@
-			COOLDOWN_START(src, rsd_scan_cooldown, RSD_ATTEMPT_COOLDOWN)
-			return FALSE
+	to_chat(user, span_blue("[target_mob] has been requested to join [target_room]."))
+	if (!invite_soul(target_room, user, target_mob))
+		to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
+		return FALSE
 
-		if(!target_room.add_soul_from_ghost(target_ghost))
-			return FALSE
+	var/target_dead = (target_mob.stat == DEAD)
+	var/mob/dead/observer/ghost = target_mob.get_ghost(TRUE, TRUE)
+
+	if (target_dead && !ghost)
+		target_mob.ghostize(TRUE)
+		ghost = target_mob.get_ghost(TRUE, TRUE) // ghostized, new ghost could be here
+
+	if (!target_mob.can_join_soulcatcher_room(target_room, TRUE)) // sanity
+		return FALSE
 
-		if(!target_mob.GetComponent(/datum/component/previous_body))
+	var/turf/source_turf = get_turf(user)
+	var/admin_log = "[key_name(user)] used [src] to put [key_name(target_mob)]'s mind into a soulcatcher at [AREACOORD(source_turf)]."
+	if (ghost)
+		if(!target_room.add_soul_from_ghost(ghost))
 			return FALSE
+	else
+		admin_log += " [target_mob] was still alive."
+		target_room.add_soul(target_mob.mind, TRUE)
 
-		var/turf/source_turf = get_turf(user)
-		log_admin("[key_name(user)] used [src] to put [key_name(target_mob)]'s mind into a soulcatcher at [AREACOORD(source_turf)]")
-		linked_soulcatcher.scan_body(target_mob, user)
-		return TRUE
+		playsound(src, 'modular_skyrat/modules/modular_implants/sounds/default_good.ogg', 50, FALSE, ignore_walls = FALSE)
+		visible_message(span_notice("[src] beeps: [target_mob]'s mind transfer is now complete."))
 
-	var/datum/soulcatcher_room/target_room = tgui_input_list(user, "Choose a room to send [target_mob]'s soul to.", name, linked_soulcatcher.soulcatcher_rooms, timeout = 30 SECONDS)
-	if(!target_room)
+	log_admin(admin_log)
+
+	if(!target_mob.GetComponent(/datum/component/previous_body))
 		return FALSE
+	linked_soulcatcher.scan_body(target_mob, user)
 
-	SEND_SOUND(target_mob, 'sound/misc/notice2.ogg')
-	window_flash(target_mob.client)
+	return TRUE
 
-	if((tgui_alert(target_mob, "Do you wish to enter [target_room]? This will remove you from your body until you leave.", name, list("Yes", "No"), 30 SECONDS, FALSE) != "Yes") || (tgui_alert(target_mob, "Are you sure about this?", name, list("Yes", "No"), 30 SECONDS, FALSE) != "Yes"))
-		COOLDOWN_START(src, rsd_scan_cooldown, RSD_ATTEMPT_COOLDOWN)
-		to_chat(user, span_warning("[target_mob] doesn't seem to want to enter."))
+/**
+ * Invites target_mob into target_room by giving them a tgui_alert.
+ * target_mob cannot be invited more than once while a alert from a given soulcatcher is open, enforced by
+ * src.confirming_entry += real_target and a check surrounding that.
+ *
+ * Args:
+ * datum/soulcatcher/target_room: The room we are inviting target_mob to.
+ * mob/living/user (Nullable): The individual who invited the mob.
+ * mob/living/target_mob: The mob that was clicked on/invited into the room.
```

Comment:
Doing it like this makes it into a bullet-point list, which is far more readable in the tooltips.
```suggestion
 * * datum/soulcatcher/target_room: The room we are inviting target_mob to.
 * * mob/living/user (Nullable): The individual who invited the mob.
 * * mob/living/target_mob: The mob that was clicked on/invited into the room.
```